frontend/src/metabase/visualizations/components/ChartSettings/BaseChartSettings/hooks.ts

```diff
@@ -17,9 +17,17 @@ export const useChartSettingsSections = ({
   shouldShowChartTypeSelector: boolean;
 }) => {
   const [currentSection, setCurrentSection] = useState<string | null>(
-    initial?.section ?? null,
+    initial?.section ??
+      (shouldShowChartTypeSelector ? DEFAULT_TAB_PRIORITY[0] : null),
   );
 
+  useEffect(() => {
+    // it's a way to select some specific tab
+    if (initial?.section) {
```

Comment:
you might want to check that there is no current section. Also in this case it would be better to avoid effects at all:
```
if (!currentSection && initial?.section) {
  setCurrentSection(initial.section);
}
```

Reply:
we basically need to update the state of currentSection from another component without redux, AFAIK this is exactly why `.initial` was introduced and I'm trying to abuse it to be able to set the tab. 

I'll double check how to optimize this code, maybe it needs to be stateful

Reply:
I didn't mean to move it anywhere, you could just replace `useEffect` with a in-render `setState` call

Reply:
This is optional, consider it a nit